### PR TITLE
Adding os filter to task grid filter.

### PR DESCRIPTION
**Pull request by codebird**
Closes https://github.com/flutter/flutter/issues/81062

I added the OSs checkboxes, and updated the filtering accordingly.

### CODE REVIEW

File: dashboard/lib/logic/task_grid_filter.dart [hunk 1]
```diff
@@ -80,11 +88,35 @@ class TaskGridFilter extends FilterPropertySource {
...
+  /// The [showWindows] property is a boolean that indicates whether to display tasks produced
+  /// by a Windows stage in the devicelab.
   /// This property will filter out columns on the build dashboard.
-  bool? get showLuci => _luciProperty.value;
-  set showLuci(bool? value) => _luciProperty.value = value;
+  bool? get showWindows => _windowsPorperty.value;
+  set showWindows(bool? value) => _windowsPorperty.value = value;
+
+  /// The [showMac] property is a boolean that indicates whether to display tasks produced
+  /// by a Windows stage in the devicelab.
```
Comment (CaseyHillers) [inline dashboard/lib/logic/task_grid_filter.dart:98]: nit: Dart docs are expected to be one sentence, empty line, then the rest of the docs.

See https://dart.dev/guides/language/effective-dart/documentation

File: dashboard/lib/logic/task_grid_filter.dart [hunk 2]
```diff
@@ -101,19 +133,27 @@ class TaskGridFilter extends FilterPropertySource {
...
-        StageName.luci: _luciProperty.fieldName,
-      };
-
   /// Check the values in the [QualifiedTask] for compatibility with the properties of this
   /// filter and return [true] iff the commit column should be displayed.
   bool matchesTask(QualifiedTask qualifiedTask) {
     if (!_taskProperty.matches(qualifiedTask.task!)) {
       return false;
     }
+    Map<String, String> showOSs = {
```
Comment (CaseyHillers) [inline dashboard/lib/logic/task_grid_filter.dart:142]: nit: Make this const

File: dashboard/lib/logic/task_grid_filter.dart [hunk 3]
```diff
@@ -101,19 +133,27 @@ class TaskGridFilter extends FilterPropertySource {
...
+      "showMac": "mac",
```
Comment (CaseyHillers) [inline dashboard/lib/logic/task_grid_filter.dart:143]: Prefer single quotes.

While at it, can you add https://dart-lang.github.io/linter/lints/prefer_single_quotes.html to the analysis_options.yaml? It seems this got dropped when moving over the recommended dart lints

File: dashboard/test/logic/task_grid_filter_test.dart [hunk 4]
```diff
@@ -140,19 +140,19 @@ void main() {
...
+    expect(trueFilterMap.matchesTask(QualifiedTask.fromTask(Task()..builderName = taskName)), true);
 
-    expect(falseFilter.matchesTask(QualifiedTask.fromTask(Task()..stageName = stageName)), false);
-    expect(falseFilterMap.matchesTask(QualifiedTask.fromTask(Task()..stageName = stageName)), false);
+    expect(falseFilter.matchesTask(QualifiedTask.fromTask(Task()..builderName = taskName)), false);
+    expect(falseFilterMap.matchesTask(QualifiedTask.fromTask(Task()..builderName = taskName)), false);
   }
 
-  test('matches Luci stage', () {
+  test('matches Windows stage', () {
```
Comment (CaseyHillers) [inline dashboard/test/logic/task_grid_filter_test.dart:150]: For tests like this, I prefer writing them with a table driven design. Something like:

```dart
for (String platform in platforms) {
  test('matches $platform', () {
    testStage(...); // This gets somewhat tricky, and you likely need to do make [platform] a map to have all the values
  });
}
```

File: dashboard/lib/build_dashboard_page.dart [hunk 5]
```diff
@@ -111,7 +111,7 @@ class BuildDashboardPageState extends State<BuildDashboardPage> {
           child: FocusTraversalGroup(
             child: SizedBox(
               width: 500,
-              height: 350,
+              height: 360,
```
Comment (CaseyHillers) [inline dashboard/lib/build_dashboard_page.dart:114]: Based on the goldens, it seems like we should add some extra height. Can you confirm it looks fine?

Comment (codebird) [reply dashboard/lib/build_dashboard_page.dart:114]: I do confirm the 10px are needed, or the the bottom buttons will be cut off. I confirm this looks fine.

File: dashboard/test/logic/task_grid_filter_test.dart (hunk 4 above)
Comment (codebird) [reply dashboard/test/logic/task_grid_filter_test.dart]: Done but I am not happy how I am setting the taskgridfilter, as it is the same way inside testStage. What do you think?

**Review by CaseyHillers — APPROVED** [review]
Thanks for the contribution!

flutter-lgtm

File: dashboard/test/logic/task_grid_filter_test.dart (hunk 4 above)
Comment (codebird) [reply dashboard/test/logic/task_grid_filter_test.dart]: I'll leave it, since you approved the PR and the test is functional, just weird looking :)

**Review by keyonghan — APPROVED** [review]
RSLGTM.

**Pull request merged**